libraries/AP_HAL_SITL/sitl_gps.cpp

```diff
@@ -330,8 +330,8 @@ void SITL_State::_update_gps_ubx(const struct gps_data *d, uint8_t instance)
     pos.latitude  = d->latitude * 1.0e7;
     pos.altitude_ellipsoid = d->altitude * 1000.0f;
     pos.altitude_msl = d->altitude * 1000.0f;
-    pos.horizontal_accuracy = 1500;
-    pos.vertical_accuracy = 2000;
+    pos.horizontal_accuracy = _sitl->gps_accuracy[instance]*1000;
+    pos.vertical_accuracy = _sitl->gps_accuracy[instance]*1000;
```

Comment:
Vertical should basically always be worse then horizontal if we want to approximate realistic GPS values. Not a deal breaker either way, as these have always been well and truly off in SITL.

Reply:
I didn't think it was worth separate params, and in the logs I was dealing with the GPS gave 1cm for both horizontal and vertical, which is what led to the issue in the EKF

Reply:
It's usually more a multiplier (IE 50% to 100% more), I was more suggesting just multiplying by 0.5 here, but not to fussed as was noted these have always been off.